Amazon.IonDotnet/Internals/Tree/SystemTreeReader.cs

```diff
@@ -154,17 +154,18 @@ public IEnumerable<SymbolToken> GetTypeAnnotationSymbols()
         }
 
         public bool HasAnnotation(string annotation)
-        {
-            if (annotation == null)
-            {
-                throw new ArgumentNullException(nameof(annotation));
-            }
-
+        {
             int? symbolTokenId = null;
             foreach (SymbolToken symbolToken in _current.GetTypeAnnotationSymbols())
             {
                 string text = symbolToken.Text;
-                if (text == null)
+
+                //zero symbol scenario
+                if (annotation == null && text == null && symbolToken.Sid == 0)
+                {
+                    return true;   
+                }
+                else if (text == null)
```

Comment:
Nit: the text == null check exists twice here, so this can be reordered to be a bit better.

Reply:
Addressed